### DISCUSSION TITLE
Need help to integrate my EgloPanel

### DISCUSSION THREAD

**Discussion by Raphinity** [fsaris/EspHome-AwoX-BLE-mesh-hub#51] [Q&A] [answered]
Hi,

I'm trying to integrate a LED Panel "Eglo 30X30".
There are 2 devices integrated in my AwoX Smart Control App : this "Eglo panel" and the "Eglo remote".

I have flashed the ESP32 ([ESP32-DevKitC-32 ESP-WROOM-32 / CP2102](https://fr.aliexpress.com/item/1005004887573569.html)) from this config : 
[awox-ble-mesh-hub.txt](https://github.com/fsaris/EspHome-AwoX-BLE-mesh-hub/files/13908013/awox-ble-mesh-hub.txt)

The logs from the device in ESPHome : 
[logs_awox-ble-mesh-hub_logs.txt](https://github.com/fsaris/EspHome-AwoX-BLE-mesh-hub/files/13908045/logs_awox-ble-mesh-hub_logs.txt)
There is nothing more in enabling verbose logging.

I have tried several times to flash the ESP32, factory reset the Led Panel and reintegrated it in my AwoX app...
I still get the same result in logs.

I don't know what I could try. Any ideas ?

Thanks !

**Comment by fsaris** [comment]
Enable debug that probably will show more

```
 debug:
   update_interval: 5s
```

**Reply by Raphinity** [reply]
_awox-ble-mesh-hub.yaml_ (ESPHome) :
```
debug:
  update_interval: 5s

# Enable logging
logger:
  level: VERY_VERBOSE
```
Home Assistant API is disabled (lines for "api" commented) -> to prevent frequent MQTT disconnections.
If it's enabled (not commented), the HA API can't connect.

_configuration.yaml_ (home assistant) :
```
# Logs
logger:
  default: warning
  logs:
    homeassistant.components.mqtt: debug
```

Logs from install (ESPHome) : 
[logs_awox-ble-mesh-hub_run.txt](https://github.com/fsaris/EspHome-AwoX-BLE-mesh-hub/files/13924834/logs_awox-ble-mesh-hub_run.txt)
-> nothing happens after "_INFO Connected to MQTT broker!_".

Mosquitto broker's logs : 
```
2024-01-13 00:41:19: New connection from 172.30.32.1:33212 on port 1883.
2024-01-13 00:41:19: New client connected from 172.30.32.1:33212 as auto-FBAB9F57-F361-7B56-CAE9-F6908E8BA66B (p2, c1, k60, u'mqtt-user-****').
2024-01-13 00:41:25: Client auto-FBAB9F57-F361-7B56-CAE9-F6908E8BA66B closed its connection.
```

**Reply by fsaris** [reply]
> If it's enabled (not commented), the HA API can't connect.

Do you have network issues?

The log from esphome should at least show some logs about the wifi connection etc.
The logs you show is only from esphome dashboard not from the device. That can take a few seconds but should show. What if you try to view the logs when connected over USB?

Can you flash some getting started esphome yaml and check if you see more logging in esphome?

**Reply by Raphinity** [reply, marked as answer]
Thank you for the tip about connecting over USB ! :)
It was clearly the MQTT connection which failed.
It was due to the _mqtt broker_ set in _awox-ble-mesh-hub.yaml_ --> simply set the IP, instead of "core-mosquitto". 

-----------------
_**Edit** : The issue described below is not relevant, as the EgloPanel was already correctly integrated.
It was a misanderstanding (read the last posts)._

Now, my _EgloPanel 30X30_ is displayed in MQTT devices, as a simple switch (functional).
In _awox-ble-mesh-hub.yaml_, I have commented the part about _device_info_ (set from your _Config examples known devices_).
```
#  device_info:
#    - product_id: 0x25
#      device_type: RGB
#      name: EGLOPanel 30X30
#      model: EPanel_300
#      manufacturer: EGLO
```
After installing again, it created a new _Unknown device type_ in MQTT, still as a simple switch.

In logs from _AwoX BLE mesh hub_ (ESPHome) :
```
[11:12:23][V][mesh_device:792]: Found existing mesh_id: 18955, Number of found mesh devices = 1
[11:12:23][D][mesh_device:395]: MAC report, dev [18955]: productID: 0x25 (Unknown device type - ) mac: A4:C1:**:**:**:**
[...]
[11:12:23][D][mesh_device:382]: status report: mesh: 18955, on: 0, color_mode: 0, transition_mode: 0, w_b: 37, temp: 101, c_b: 29, rgb: FFFFFF
```

Is there a way to "recognize" it ? Which informations from logs would help ?

Thanks again !

**Reply by fsaris** [reply]
Can you show the logs with the device info enabled?

```
#  device_info:
#    - product_id: 0x25
#      device_type: RGB
#      name: EGLOPanel 30X30
#      model: EPanel_300
#      manufacturer: EGLO
```

And are you sure the light isn't also recognized by HA? There should be a similar mqtt message like you found for the switch but then with `light` instead of `switch` in the topic.
Check also the HA logs if there isn't a message about issues with processing the discovery message

**Reply by Raphinity** [reply]
Parts from logs (after installing with the device info enabled) :
[logs_awox-ble-mesh-hub_run (parts).txt](https://github.com/fsaris/EspHome-AwoX-BLE-mesh-hub/files/13928260/logs_awox-ble-mesh-hub_run.parts.txt)

The HA logs level is "warning" and it doesn't display any issue with processing the discovery message 

In HA devices list :
![image](https://github.com/fsaris/EspHome-AwoX-BLE-mesh-hub/assets/146590746/7777280d-4ec6-4120-8726-ef742a045297)

In MQTT (Mosquitto broker integrated to Home Assistant) :
![image](https://github.com/fsaris/EspHome-AwoX-BLE-mesh-hub/assets/146590746/0e20b81d-40b3-4c7c-9e33-b0435135df2f)

**Reply by fsaris** [reply]
So the light is recognized and online. Can't you control it from HA? Or does is work now?

**Reply by Raphinity** [reply]
Alright ! I misunderstood it...
It's displayed as a simple switch in MQTT, but it is fully integrated as a RGB Led panel.

So, I can use the "_Hue-Like Light Card_" (available from HACS) and it is  fully controllable now.
Really good ! Thank you for this great work, and your help to integrate my EgloPanel ! 👍
